cypress/integration/po/pages/first-login.po.ts

```diff
@@ -0,0 +1,53 @@
+import PagePo from '@/cypress/integration/po/pages/page.po';
+import LabeledInputPo from '@/cypress/integration/po/components/labeled-input.po';
+import CheckboxInputPo from '@/cypress/integration/po/components/checkbox-input.po';
+import RadioGroupInputPo from '@/cypress/integration/po/components/radio-group-input.po';
+import AsyncButtonPo from '@/cypress/integration/po/components/async-button.po';
+import FormPo from '@/cypress/integration/po/components/form.po';
+
+export class FirstLoginPagePo extends PagePo {
+  static url: string = '/auth/login'
+  static goTo(): Cypress.Chainable<Cypress.AUTWindow> {
+    return super.goTo(FirstLoginPagePo.url);
+  }
+
+  form: FormPo;
+
+  constructor() {
+    super(FirstLoginPagePo.url);
+
+    this.form = new FormPo('form', this.self());
+  }
+
+  choosePassword(): RadioGroupInputPo {
+    return new RadioGroupInputPo(this.form.radioGroups().eq(0));
```

Comment:
I meant the components within the page (if radio buttons/labels are added or removed then these indexes will point to the wrong ones)

Reply:
This is what I was trying to ask you from the beginning, about the use of components PO in other pages too, where we do not mention the ID but rather use identifier of this type:

```ts
  username(): LabeledInputPo {
    return new LabeledInputPo(this.form.labels().first());
  }

  password(): LabeledInputPo {
    return new LabeledInputPo(this.form.labels().eq(1));
  }
```

Should we replace it everywhere?

Reply:
In master `first` and `eq` are only used in the login page po and a lot in the old product-side-nav.spec (which hadn't been updated). It was a bad pattern to start with, I don't think we should compound it by continuing to use it. Lets concentrate on using the new common pattern of data attribute > id > classes

Reply:
Totally agree, let's do it! 👍

Reply:
For some reasons seems to do not work like this:
```ts
  password(): LabeledInputPo {
    return new LabeledInputPo(cy.find('[data-testid="local-login-password"]'));
  }
  ```

As I get the [following error](https://github.com/rancher/dashboard/runs/6554230671?check_suite_focus=true#step:6:69):

![Screenshot 2022-05-23 at 13 54 08](https://user-images.githubusercontent.com/5009481/169813780-752391d6-5005-41a6-9334-e58e2fe0ab9e.png)

We may rethink the use of PO for the components at this point, as it may be more convenient something like
```
    cy.find('[data-testid="local-login-password"]')
       .setIntup(Cypress.env('bootstrapPassword'));
```
Instead of creating 2 POs and have uncommon CY structure.

Reply:
Getting in some trouble while applying the ID to the components. In the case of the LabelInput this is passed directly to the input, while for the Password component it's not.

Both of them share however the same PO, which results incorrect in this case.